axelor-production/src/main/java/com/axelor/apps/production/web/ManufOrderController.java

```diff
@@ -459,4 +460,45 @@ public void computeCostPrice(ActionRequest request, ActionResponse response) {
       TraceBackService.trace(response, e);
     }
   }
+
+  public void checkMergeValues(ActionRequest request, ActionResponse response) {
+    try {
+      if (request.getContext().get("id") != null) {
+        response.setError(I18n.get("Please select more than one OF"));
+      } else {
+        Object _ids = request.getContext().get("_ids");
+        if (!ObjectUtils.isEmpty(_ids)) {
+          List<Long> ids = (List<Long>) _ids;
+          if (ids.size() < 2) {
+            response.setError(I18n.get("Please select more than one OF"));
+          } else {
+            boolean canMerge = Beans.get(ManufOrderWorkflowService.class).canMerge(ids);
+            if (canMerge) {
+              response.setAlert(I18n.get("Are you sur you want to merge those manuf orders?"));
+            } else {
+              response.setError(
+                  I18n.get(
+                      "The merge must concerns only manuf order with Draft or Planned status, with the same products and same workshop stock location and will all bill of material compatibles."));
+            }
+          }
+        } else {
+          response.setError(I18n.get("Please select something to validate"));
+        }
+      }
+
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void generateMergeManufOrder(ActionRequest request, ActionResponse response) {
+    try {
+      Object _ids = request.getContext().get("_ids");
```

Comment:
Is the case where no line is selected handled ? Either here or in the "merge" method, I don't see it when looking rapidly.